lib/ledger.js

```diff
@@ -99,147 +97,278 @@ async function _findAgent() {
+  const adminAccount = {
+    id,
+    email: 'admin@' + config.server.domain,
+  };
+  const meta = {
+    sysResourceRole: [{
+      sysRole: 'veres.admin',
+      generateResource: 'id'
+    }],
+  };
+
+  await brAccount.insert({actor: null, account: adminAccount, meta});
+  // the admin account exists now
+  return brAccount.getCapabilities({id});
+}
+
+async function _getEndpoints({hostnames}) {
+  return Promise.all(hostnames.map(hostname => (async () => {
+    const client = new WebLedgerClient(
+      {hostname, httpsAgent: brHttpsAgent.httpsAgent});
+
+    try {
+      // there is no benefit in running getServiceEndpoint and getTargetNode
+      // in parallel
+      const endpoint = await client.getServiceEndpoint(
+        {serviceId: 'ledgerOperationService'});
+      const targetNode = await client.getTargetNode();
+      return {client, endpoint, hostname, targetNode};
+    } catch(e) {
+      // FIXME: remove overly verbose axios error, WebLedgerClient should be
+      // update to tame these errors
+      delete e.details.error;
+      throw e;
     }
-    // create admin identity if it doesn't exist
-    const adminAccount = {
-      id,
-      email: 'admin@' + config.server.domain,
-    };
-    const meta = {
-      sysResourceRole: [{
-        sysRole: 'veres.admin',
-        generateResource: 'id'
-      }],
-    };
-    brAccount.insert(
-      {actor: null, account: adminAccount, meta}, err => {
-        if(err) {
-          return callback(err);
-        }
-        brAccount.getCapabilities({id}, callback);
-      });
+  })()));
+}
+
+async function _sendElectorPool({electorPoolId, maintainerDoc}) {
+  const method = maintainerDoc.getVerificationMethod(
+    {proofPurpose: 'capabilityInvocation'});
+  const capabilityInvocationKey = maintainerDoc.keys[method.id];
+  const hostnames = config['veres-one'].electorHosts;
+  let targets;
+  try {
+    targets = await pRetry(() => _getEndpoints({hostnames}), {
+      retries: 300,
+      onFailedAttempt: async error => {
+        logger.error(
+          'Unable to acquire details from electorHosts. Retrying...', {error});
+        await delay(5000);
+      }
+    });
+  } catch(e) {
+    logger.debug('Failed to collect target details.', {errorName: e.name});
+    throw e;
+  }
+
+  const didContexts = [
+    constants.DID_CONTEXT_URL,
+    constants.VERES_ONE_CONTEXT_V1_URL
+  ];
+  // targets = [{endpoint, hostname, targetNode}]
+  const electorPoolDoc = {
+    '@context': didContexts,
+    id: electorPoolId,
+    type: 'ElectorPool',
+    // FIXME: this has to be in the v1 context before we can sign documents
+    // veresOneTicketRate: 10, /* TBD */
+    controller: maintainerDoc.doc.id,
+    electorPool: [],
+  };
+  for(const t of targets) {
+    electorPoolDoc.electorPool.push({
+      // FIXME: is this right? where does the elector nym did come from?
+      id: `urn:uuid:${uuid()}`,
+      elector: 'did:v1:nym:z6MkgxjunEaHCzuu56Nkgj6WdCQG8VhY8jyRYEFgmQCXF4Rp',
```

Comment:
Confused here... why is the elector hard coded? Why isn't it fetched from each elector host? Are we not checking this?

Reply:
At present, electors do not have DIDs.  We had some data model at some point that said there is an `elector` property that is a DID.  The confusion here is reflected by the FIXME on L334.  I don't know the answer.

Reply:
Why is there just one elector ID? I'm very confused by this. Do we need this property at all right now?